### PR TITLE
Syn library update

### PR DESCRIPTION
**Pull request by mihaicalinluca**

### CODE REVIEW

File: framework/base/src/types/managed/wrapped/egld_or_multi_esdt_payment.rs [hunk 1]
```diff
@@ -16,11 +16,70 @@ use crate::derive::TypeAbi;
...
-    TopDecode, TopEncode, NestedDecode, NestedEncode, TypeAbi, Clone, PartialEq, Eq, Debug,
+    TopDecode, TopEncode, TypeAbi, NestedDecode, NestedEncode, Clone, PartialEq, Eq, Debug,
 )]
 pub enum EgldOrMultiEsdtPayment<M: ManagedTypeApi> {
     Egld(BigUint<M>),
     MultiEsdt(ManagedVec<M, EsdtTokenPayment<M>>),
 }
 
 impl<M> CodecFromSelf for EgldOrMultiEsdtPayment<M> where M: ManagedTypeApi {}
+
```
Comment (andrei-marinica) [inline framework/base/src/types/managed/wrapped/egld_or_multi_esdt_payment.rs:27]: clean up

File: contracts/examples/adder/wasm/src/lib.rs [hunk 2]
```diff
@@ -22,9 +22,8 @@ multiversx_sc_wasm_adapter::endpoints! {
     adder
     (
         init => init
-        getSum => sum
-        upgrade => upgrade
-        add => add
+        endpoint => try_s
```
Comment (andrei-marinica) [inline contracts/examples/adder/wasm/src/lib.rs:25]: call `sc-meta all abi` again

File: contracts/examples/adder/src/adder.rs [hunk 3]
```diff
@@ -1,7 +1,6 @@
 #![no_std]
 
 multiversx_sc::imports!();
-
```
Comment (andrei-marinica) [inline contracts/examples/adder/src/adder.rs:4]: revert

File: framework/derive/src/macro_contract.rs [hunk 4]
```diff
@@ -8,9 +8,22 @@ pub fn process_contract(
     input: proc_macro::TokenStream,
 ) -> proc_macro::TokenStream {
     let new_input = trait_preprocessing(input);
-    let args_input = parse_macro_input!(args as syn::AttributeArgs);
     let proc_input = &parse_macro_input!(new_input as syn::ItemTrait);
 
+    let args_input = if args.is_empty() {
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply framework/derive/src/macro_contract.rs:13]: Also, the entire parsing should go into `parse_contract_trait`. De-duplicates a few lines.
`parse_contract_trait` should just take the raw token streams.

**Review by andrei-marinica — APPROVED** [review]

**Review by alyn509 — APPROVED** [review]

**Review by BiancaIalangi — APPROVED** [review]

**Pull request merged**